src/review/commands/build.ts

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Command, flags } from '@oclif/command';
+import * as fg from 'fast-glob';
+import * as fileSize from 'filesize';
+import { promises, statSync } from 'fs';
+import * as Path from 'path';
+import {
+  getPackageJsonDeps,
+  getPackageLockDeps,
+  nodeModulesGlobs,
+} from '../bundle';
+import { compileJsModule } from '../bytecode';
+import { Entry, write } from '../format';
+
+// tslint:disable-next-line: no-default-export
+export default class Build extends Command {
+  static description = 'build a bundle';
+  static examples = [`$ bulkan build`];
+
+  static flags = {
+    help: flags.help({ char: 'h' }),
+    no_compile: flags.boolean({
+      char: 'C',
+      description: "don't compile javascript files",
+    }),
+    outfile: flags.string({
+      char: 'o',
+      description: 'output file for the generated bundle.',
+      default: 'bundle.blkn',
+    }),
+  };
+
+  static args = [{ name: 'cwd', default: './' }];
+
+  async run() {
+    const { args, flags } = this.parse(Build);
+
+    const handlers: { [extension: string]: Handler } = {
+      '.json': loadBuffer,
+      '.js': loadBufferAndCompileCache,
+    };
+    if (flags.no_compile) handlers['.js'] = loadBuffer;
+
+    this.log('Collecting paths');
+    const pkgPath = Path.resolve(Path.join(args.cwd, 'package.json'));
+    const pkgLockPath = Path.resolve(Path.join(args.cwd, 'package-lock.json'));
+    const pkgDeps = getPackageJsonDeps(require(pkgPath));
+    const pkgLockDeps = getPackageLockDeps(require(pkgLockPath));
+    const globs = nodeModulesGlobs([...new Set([...pkgDeps, ...pkgLockDeps])]);
+
+    const paths = await fg(globs);
+    this.log(`Found ${paths.length} paths`);
+
+    this.log(`Preparing buffers`);
+    const entryPs = ([] as Array<Promise<Entry>>)
+      .concat(
```

Comment:
I think this can be simplified with [flatMap](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap):

```
paths.flatMap((path: string) => {
  const ext = Path.extname(path);
  if (ext in handlers) return handlers[ext](path);
  return [];
})
```

Reply:
Also, the if statement may go away if a `filter` is done before the `map`/`flatMap`.